File: Composite/Dart/Group.dart
import 'IUser.dart';

class Group implements IUser {
  List<IUser> _users = [];

  void add(IUser user) {
    _users.add(user);
  }

  @override
  void getUserDetails() {
    for (var user in _users) {
      user.getUserDetails();
    }
  }
}

File: Composite/Javascript/Group.js
import IUser from "./IUser.js";

class Group extends IUser {
  constructor() {
    super();
    this.users = [];
  }

  add(user) {
    this.users.push(user);
  }

  inspect() {
    this.users.forEach(user => {
      user.inspect();
    });
  }
}

export default Group;